Precise nodejs version, how to switch and node-gyp dependency on Windows (#1055)
PrestaShop/docs: development/compile-assets.md

## Requirements

> We use [Webpack](https://webpack.js.org/) to compile assets. PrestaShop is using NodeJS ([get it here](https://nodejs.org/)), NPM will take care of it all.

Reviewer: ```suggestion
PrestaShop uses [Webpack](https://webpack.js.org/) to compile assets, which requires NodeJS ([get it here](https://nodejs.org/)).
```

> Here is a list of compatibility:

Reviewer: You will need a different NodeJS version depending on the PrestaShop version you're working with:

| PrestaShop Versions | NodeJS Versions |
|---------------------|-----------------|

> | under 1.7.7         | 8.x and older   |

Reviewer: ```suggestion
| 1.7.0~1.7.6          | 8.x   |
```

| 1.7.7               | 10.x            |

> | 1.7.8               | 12.x and 14.x   |

Reviewer: We are still using node 10 for sanity tests (see https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/.github/workflows/sanity-74.yml#L39). Is it something that we should fix?

Also, what is the recommended version?

Author: As 1.7.8.x is compatible with latest LTS, I think we should make tests compatibles with the latest LTS too

We should use the same version everywhere, overwise it's a nightmare to use

> If you can't manage to compile assets, you may be using an older version of PrestaShop. We strongly recommend you ton install a package such as ['n'](https://www.npmjs.com/package/n?activeTab=versions) or [nvm](https://github.com/nvm-sh/nvm) to try with an older version of NodeJS.

Reviewer: I suppose you meant this:
```suggestion
If you are struggling to compile assets, you might be using the wrong NodeJs version. We strongly recommend you to use a version manager such as ['n'](https://www.npmjs.com/package/n) or [nvm](https://github.com/nvm-sh/nvm) so you can easily switch between NodeJS versions.
```

> On Windows, you'll probably need to install `windows-build-tools` using `npm i --global windows-build-tools` in order to make [node-gyp](https://github.com/nodejs/node-gyp#on-windows) working.

Reviewer: ```suggestion
On Windows, you'll probably need to install `windows-build-tools` using `npm i --global windows-build-tools` in order to [make node-gyp work](https://github.com/nodejs/node-gyp#on-windows).
```
